src/main/java/bit/javaoop/AbstractEmployee.java

```diff
@@ -26,4 +26,7 @@ public double getSalary() {
     public void setSalary(double salary) {
         this.salary = salary;
     }
+
+    abstract public String toString(int line);
```

Comment:
Ciekawy pomysł.

Nie mówię, że zły, jednak chcieliśmy żeby to `Company` było odpowiedzialne za sparsowanie i wyświetlenie hierarchii pracowników. Twój pomysł też jest dobry, ale ma istotną wadę. Mianowicie chcąc zmienić sposób wyświetlania hierarchii pracowników musielibyśmy to zrobić w kilku miejscach, a nie tak jak w naszym rozwiązaniu - tylko w jednym. 

Ponadto, jest jeszcze jedna sprawa. Załóżmy pewien scenariusz. Chcemy dodać do firmy stanowisko dajmy na to Biznesmen (tak wiem okropna nazwa, nie mam kompletnie pomysłów xD). Biznesmeni dziedziczą po `AbstractEmployee`. I teraz muszą zaimplementować metodkę `toString(int line)`. I mamy powtórzenie kodu.

Z takich drobiazgów, to nie za bardzo wiadomo o co chodzi ze zmienną `line`. Mogłaby się nazywać np. `indentDepth` - o ile dobrze zrozumiałem jej kontekst.